ydb/core/data_integrity_trails/data_integrity_trails.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+namespace NKikimr {
+namespace NDataIntegrity {
+
+inline void LogKeyValue(const TString& key, const TString& value, TStringStream& ss, bool last = false) {
+    ss << key << ": " << (value.Empty() ? "Empty" : value) << (last ? "" : ",");
+}
+
+template <class TransactionSettings>
+inline void LogTxSettings(const TransactionSettings& txSettings, TStringStream& ss) {
+    switch (txSettings.tx_mode_case()) {
+        case TransactionSettings::kSerializableReadWrite:
+            LogKeyValue("TxMode", "SerializableReadWrite", ss);
+            break;
+        case TransactionSettings::kOnlineReadOnly:
+            LogKeyValue("TxMode", "OnlineReadOnly", ss);
```

Comment:
Hmm, are we logging reads as well? I thought we are only interested in writes.

Reply:
Query type is undefined on GRPC level, so we should  log all requests here. But we use different levels of logging for executer/session_actor/grpc:

grpc - DEBUG (all requests)
session_actor - INFO (requests that can potentially contain modifications )
executer - NOTICE (requests with modifications)